File: 0242-valid-anagram/0242-valid-anagram.cpp
class Solution {
public:
    bool isAnagram(string s, string t) {
        vector<int> sarr(26,0);
        vector<int> tarr(26,0);
        if(s.length()!=t.length())return false;
        for(int i=0;i<s.length();i++){
            sarr[s[i]-'a']++;
            tarr[t[i]-'a']++;
        }
        for(int i=0;i<26;i++){
            if(sarr[i]!=tarr[i])return false;
        }
        return true;
    }
    
};

File: 0242-valid-anagram/0242-valid-anagram.java
class Solution {
    public boolean isAnagram(String s, String t) {
//         int[] arrs = new int[26];
//         int[] arrt = new int[26];
//         if(s.length()!=t.length())return false;

//         for(int i=0;i<s.length();i++){
//             arrs[s.charAt(i)-'a']++;
//             arrt[t.charAt(i)-'a']++;
//         }
//         for(int i=0;i<26;i++){
//             if(arrs[i]!=arrt[i])return false;
//         }
//         return true;
        
        
        char[] sChars = s.toCharArray();
        char[] tChars = t.toCharArray();
        
        Arrays.sort(sChars);
        Arrays.sort(tChars);
        
        return Arrays.equals(sChars, tChars);


    }
}